test/fixtures/privacy-setting-extension/content.js

```diff
@@ -0,0 +1,48 @@
+test("privacy API should be unavailable in the content script", (t) => {
+  t.deepEqual(browser.privacy, undefined, "browser.privacy is not available in a content script");
+});
+
+test("privacy API should support promises", async (t) => {
+  async function callSettingAPI(method, ...args) {
+    // Invokes: browser.privacy.services.passwordSavingEnabled[method](...args);
+    let {type, result} = await browser.runtime.sendMessage({method, args});
+    // In Chrome `undefined` values are serialized to `null`, so use the
+    // pre-determined type instead.
+    return type === "undefined" ? undefined : result;
+  }
+
+  let res = await callSettingAPI("get", {});
+  t.deepEqual(res, {
+    levelOfControl: "controllable_by_this_extension",
+    // Enabled by default in Chrome; disabled by default in Firefox.
+    value: !navigator.userAgent.includes("Firefox/"),
+  }, "passwordSavingEnabled.get() resolves to the initial value");
+
+  const defaultValue = res.value;
+  const newValue = !defaultValue;
+  res = await callSettingAPI("set", {value: newValue});
+  if (navigator.userAgent.includes("Firefox/")) {
+    t.equal(res, true, "passwordSavingEnabled.set() resolves to true");
+  } else {
+    t.equal(res, undefined, "passwordSavingEnabled.set() resolves to a void value");
```

Comment:
note to self: this different behavior of `set` and `clear` seems something that we should list in the known incompatibilities in the updated README (iirc on Firefox `set` and `clear` returns true if the setting has been changes, false otherwise, on the contrary Chrome doesn't pass any parameter to the callback that is called once the operation has been completed).

Reply:
This seems like something that should be documented on MDN: https://developer.mozilla.org/en-US/Add-ons/WebExtensions/API/types/BrowserSetting